File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        def bfs(node, low, high) -> bool:
            if not node:
                return True
            
            if not ( low < node.val < high ):
                return False
            
            return (bfs(node.left, low, node.val) and
                bfs(node.right, node.val, high)
                )
        return bfs(root, float("-inf"), float("inf"))

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
    res := make([][]int, 0)
    if root == nil {
        return res
    }

    queue := []*TreeNode{root}

    for len(queue) > 0 {
        size := len(queue)
        lvl := make([]int, 0)

        for i := 0; i < size; i++ {
            node := queue[0]
            queue = queue[1:]

            lvl = append(lvl, node.Val)

            if node.Left != nil {
                queue = append(queue, node.Left)
            }
            if node.Right != nil {
                queue = append(queue, node.Right)
            }
        }
        res = append(res, lvl)
    }
    return res
}
